### ISSUE TITLE
[HttpListener] [Porting] All calls to HttpListenerWebSocketContext.WebSocket.Receive fail with ArgumentNullException

### ISSUE THREAD

**Issue by hughbe** [dotnet/corefx#20391]
Following test code passes with netfx, fails with an ArgumentNullException in netcoreapp:
```cs
// Licensed to the .NET Foundation under one or more agreements.
// The .NET Foundation licenses this file to you under the MIT license.
// See the LICENSE file in the project root for more information.

using System.Net.WebSockets;
using System.Text;
using System.Threading;
using System.Threading.Tasks;
using Xunit;

namespace System.Net.Tests
{
    public class HttpListenerWebSocketStreamTests : IDisposable
    {
        public static bool IsNotWindows7OrUapCore { get; } = !PlatformDetection.IsWindows7 && PlatformDetection.IsNotOneCoreUAP;

        private HttpListenerFactory Factory { get; }
        private HttpListener Listener { get; }
        private ClientWebSocket Client { get; }
        private Task ClientConnectTask { get; set; }

        public HttpListenerWebSocketStreamTests()
        {
            Factory = new HttpListenerFactory();
            Listener = Factory.GetListener();
            Client = new ClientWebSocket();
        }

        public void Dispose()
        {
            Factory.Dispose();
            Client.Dispose();

            if (ClientConnectTask?.IsCompleted == true)
            {
                ClientConnectTask?.Dispose();
            }
        }

        [ConditionalFact(nameof(IsNotWindows7OrUapCore))]
        public async Task ReceiveAsync_ReadWholeBuffer_Success()
        {
            HttpListenerWebSocketContext context = await GetWebSocketContext();

            await ClientConnectTask;
            const string Text = "Hello Web Socket";
            byte[] sentBytes = Encoding.ASCII.GetBytes(Text);

            await Client.SendAsync(new ArraySegment<byte>(sentBytes), WebSocketMessageType.Text, false, new CancellationToken());

            byte[] receivedBytes = new byte[sentBytes.Length];
            WebSocketReceiveResult result = await context.WebSocket.ReceiveAsync(new ArraySegment<byte>(receivedBytes), new CancellationToken());
            Assert.Equal(WebSocketMessageType.Text, result.MessageType);
            Assert.False(result.EndOfMessage);
            Assert.Null(result.CloseStatus);
            Assert.Null(result.CloseStatusDescription);

            Assert.Equal(Text, Encoding.ASCII.GetString(receivedBytes));
        }

        [ConditionalFact(nameof(IsNotWindows7OrUapCore))]
        public async Task ReceiveAsync_EmptyInnerBuffer_ThrowsArgumentNullException()
        {
            HttpListenerWebSocketContext context = await GetWebSocketContext();
            await Assert.ThrowsAsync<ArgumentNullException>("buffer.Array", () => context.WebSocket.ReceiveAsync(new ArraySegment<byte>(), new CancellationToken()));
        }

        private async Task<HttpListenerWebSocketContext> GetWebSocketContext(string[] subProtocols = null)
        {
            if (subProtocols != null)
            {
                foreach (string subProtocol in subProtocols)
                {
                    Client.Options.AddSubProtocol(subProtocol);
                }
            }

            var uriBuilder = new UriBuilder(Factory.ListeningUrl) { Scheme = "ws" };
            Task<HttpListenerContext> serverContextTask = Factory.GetListener().GetContextAsync();

            ClientConnectTask = Client.ConnectAsync(uriBuilder.Uri, CancellationToken.None);
            if (ClientConnectTask == await Task.WhenAny(serverContextTask, ClientConnectTask))
            {
                await ClientConnectTask;
                Assert.True(false, "Client should not have completed prior to server sending response");
            }

            HttpListenerContext context = await serverContextTask;
            return await context.AcceptWebSocketAsync(null);
        }
    }
}
```

**Comment by hughbe** [comment]
Stack trace:
```
System.Net.Tests.HttpListenerWebSocketStreamTests.ReceiveAsync_ReadWholeBuffer_Success [FAIL]
        System.ArgumentNullException : Value cannot be null.
        Parameter name: array
        Stack Trace:
              at System.ThrowHelper.ThrowArgumentNullException(ExceptionArgument argument)
              at System.ArraySegment`1..ctor(T[] array)
              at System.ArraySegment`1.op_Implicit(T[] array)
              at System.Net.WebSockets.WebSocketBase.<ReceiveAsyncCore>d__45.MoveNext()
           --- End of stack trace from previous location where exception was thrown ---
              at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
              at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
              at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
              at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task task)
              at System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()
              at System.Net.Tests.HttpListenerWebSocketStreamTests.<ReceiveAsync_ReadWholeBuffer_Success>d__18.MoveNext
  ()
           --- End of stack trace from previous location where exception was thrown ---
              at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
              at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
              at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
              at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task task)
              at System.Runtime.CompilerServices.TaskAwaiter.GetResult()
           --- End of stack trace from previous location where exception was thrown ---
              at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
              at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
              at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
              at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task task)
              at System.Runtime.CompilerServices.TaskAwaiter.GetResult()
           --- End of stack trace from previous location where exception was thrown ---
              at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
              at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
              at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
              at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task task)
              at System.Runtime.CompilerServices.TaskAwaiter.GetResult()
```

**Comment by hughbe** [comment]
I have a fix

**Comment by hughbe** [comment]
This was actually fixed (not contributed to) in #20398

**Comment by karelz** [comment]
@stephentoub was it part of your rel/2.0.0 port? (if there is easy way for me to tell, please let me know, thanks!)

**Comment by stephentoub** [comment]
> was it part of your rel/2.0.0 port?

Yes.

> if there is easy way for me to tell, please let me know

@hughbe linked to the PR that fixed it: #20398.  In that PR you see three commits:
https://github.com/dotnet/corefx/pull/20398/commits
and you can see all three of those commits listed in the porting PR here:
https://github.com/dotnet/corefx/pull/20422/commits

**Comment by karelz** [comment]
Thanks!
